File: examples/javascript/fetch-incentives.js
/**
 * ZOE API - Fetch State Incentives Example (JavaScript)
 *
 * This example demonstrates fetching SGIP incentive data
 * from the ZOE free tier via jsDelivr CDN.
 */

async function fetchSGIPIncentives() {
  const url = 'https://cdn.jsdelivr.net/gh/securedcarbon/zoe-data-public@main/samples/state-incentives/sgip-current-step.json';

  try {
    const response = await fetch(url);
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const incentives = await response.json();

    console.log('SGIP Incentive Data:');
    console.log(`Program: ${incentives.program_name}`);
    console.log(`Current Rate: $${incentives.rate}/kWh`);
    console.log(`Budget Remaining: $${incentives.budget_remaining.toLocaleString()}`);
    console.log(`Eligible Sectors: ${incentives.eligible_sectors.join(', ')}`);

    return incentives;
  } catch (error) {
    console.error('Error fetching SGIP data:', error);
    throw error;
  }
}

async function fetchCensusTract(geoid) {
  const url = 'https://cdn.jsdelivr.net/gh/securedcarbon/zoe-data-public@main/samples/census/tracts-ca-sample.json';

  try {
    const response = await fetch(url);
    const tracts = await response.json();

    const tract = tracts.find(t => t.geoid === geoid);

    if (!tract) {
      console.warn(`Tract ${geoid} not found in free tier sample. Upgrade to Pro for full coverage.`);
      return null;
    }

    console.log(`Census Tract ${geoid}:`);
    console.log(`County: ${tract.county}, ${tract.state}`);

    return tract;
  } catch (error) {
    console.error('Error fetching census data:', error);
    throw error;
  }
}

// Example usage
async function main() {
  console.log('=== ZOE API Free Tier Example ===\n');

  // Fetch SGIP incentives
  const sgip = await fetchSGIPIncentives();
  console.log('\n---\n');

  // Fetch a specific census tract
  const tract = await fetchCensusTract('06037260100'); // Los Angeles tract

  if (tract && sgip) {
    console.log('\n=== Combined Analysis ===');
    console.log(`For tract ${tract.geoid} in ${tract.county}:`);
    console.log(`SGIP incentive available: $${sgip.rate}/kWh`);
    console.log('\nUpgrade to Pro for full demographic data and real-time incentive tracking!');
  }
}

// Run if executed directly
if (typeof require !== 'undefined' && require.main === module) {
  main().catch(console.error);
}

// Export for use in other modules
if (typeof module !== 'undefined' && module.exports) {
  module.exports = {
    fetchSGIPIncentives,
    fetchCensusTract
  };
}

File: examples/python/fetch_incentives.py
"""
ZOE API - Fetch State Incentives Example (Python)

This example demonstrates fetching SGIP incentive data
from the ZOE free tier via jsDelivr CDN.
"""

import requests
import json
from typing import Optional, Dict, Any


def fetch_sgip_incentives() -> Dict[str, Any]:
    """
    Fetch SGIP incentive data from ZOE free tier.

    Returns:
        dict: SGIP incentive program data
    """
    url = 'https://cdn.jsdelivr.net/gh/securedcarbon/zoe-data-public@main/samples/state-incentives/sgip-current-step.json'

    try:
        response = requests.get(url)
        response.raise_for_status()

        incentives = response.json()

        print('SGIP Incentive Data:')
        print(f"Program: {incentives['program_name']}")
        print(f"Current Rate: ${incentives['rate']}/kWh")
        print(f"Budget Remaining: ${incentives['budget_remaining']:,}")
        print(f"Eligible Sectors: {', '.join(incentives['eligible_sectors'])}")

        return incentives
    except requests.RequestException as e:
        print(f"Error fetching SGIP data: {e}")
        raise


def fetch_census_tract(geoid: str) -> Optional[Dict[str, Any]]:
    """
    Fetch census tract data from ZOE free tier.

    Args:
        geoid: 11-digit FIPS code

    Returns:
        dict: Census tract data, or None if not found in sample
    """
    url = 'https://cdn.jsdelivr.net/gh/securedcarbon/zoe-data-public@main/samples/census/tracts-ca-sample.json'

    try:
        response = requests.get(url)
        response.raise_for_status()

        tracts = response.json()

        tract = next((t for t in tracts if t['geoid'] == geoid), None)

        if not tract:
            print(f"Tract {geoid} not found in free tier sample. Upgrade to Pro for full coverage.")
            return None

        print(f"Census Tract {geoid}:")
        print(f"County: {tract['county']}, {tract['state']}")

        return tract
    except requests.RequestException as e:
        print(f"Error fetching census data: {e}")
        raise


def main():
    """Example usage of ZOE API free tier."""
    print("=== ZOE API Free Tier Example ===\n")

    # Fetch SGIP incentives
    sgip = fetch_sgip_incentives()
    print("\n---\n")

    # Fetch a specific census tract
    tract = fetch_census_tract('06037260100')  # Los Angeles tract

    if tract and sgip:
        print("\n=== Combined Analysis ===")
        print(f"For tract {tract['geoid']} in {tract['county']}:")
        print(f"SGIP incentive available: ${sgip['rate']}/kWh")
        print("\nUpgrade to Pro for full demographic data and real-time incentive tracking!")


if __name__ == '__main__':
    main()
